### PR TITLE
feat(Layout): add JustifiedLayout

### PR DESCRIPTION
**Pull request by itoolsg**
## Issue

add JustifiedLayout #32 

## Details

add JustifiedLayout
test JustifiedLayout unit

### CODE REVIEW

File: test/manual/layouts/demo.html [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+		.item .thumbnail {
+			background: transparent;
+			border: none;
+			border-radius: 0;
+			box-shadow: none;
+			padding: 0;
+		}
+	</style>
+<script>
+	var _a = new JustifiedLayout({
```
Comment (jongmoon) [inline test/manual/layouts/demo.html]: `_a` looks like not used. :)

File: src/layouts/JustifiedLayout.js [hunk 2]
```diff
@@ -0,0 +1,215 @@
...
+class JustifiedLayout {
+	constructor(options = {}) {
+		this.options = options;
+	}
+	remove(removedItem, items, outline) {
+		const direction = _style[this.options.direction] ? this.options.direction : VERTICAL;
+		const style = _style[direction];
+		const _pos1 = style[0];
+		const _pos2 = style[1];
+
```
Comment (jongmoon) [inline src/layouts/JustifiedLayout.js]: @sculove Is `underbar prefix on local variable` accord with egjs coding convention?

File: src/layouts/JustifiedLayout.js [hunk 3]
```diff
@@ -0,0 +1,215 @@
...
+function getTopPoint(outlines, pos) {
+	return Math.min(...outlines.map(outline => outline[pos]));
+}
+function getBottomPoint(outlines, pos) {
+	return Math.max(...outlines.map(outline => outline[pos]));
+}
+
+class JustifiedLayout {
+	constructor(options = {}) {
+		this.options = options;
```
Comment (sculove) [inline src/layouts/JustifiedLayout.js]: How about use default options like this?
```js
this.options = {...{ direction: VERTICAL }, ...options};
```

File: config/webpack.config.layout.js [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+var webpack = require("webpack");
+var UglifyJSPlugin = require("uglifyjs-webpack-plugin");
+var uglifyConfig = require("./uglify");
+var banner = require("./banner");
+var path = require("path");
+
+var config = {
+	entry: {
+		"JustifiedLayout": "./src/layouts/JustifiedLayout.js",
+		"GridLayout": "./src/layouts/GridLayout.js",
```
Comment (sculove) [inline config/webpack.config.layout.js]: How about changing Universal term like Layout?

File: src/layouts/JustifiedLayout.js [hunk 5]
```diff
@@ -0,0 +1,215 @@
+import dijkstra from "../../lib/dijkstra";
+
+const _style = {
```
Comment (sculove) [inline src/layouts/JustifiedLayout.js]: How about changing uppercase?
const should use uppercase type

File: src/layouts/JustifiedLayout.js [hunk 6]
```diff
@@ -0,0 +1,215 @@
...
+
+		this._layout(items, 0, items.length, point, APPEND);
+	}
+	prepend(items, outlines) {
+		const _pos1 = this.getStyleNames()[0];
+		const point = getTopPoint(outlines, _pos1);
+
+		this._layout(items, 0, items.length, point, PREPEND);
+	}
+	_layout(items, _i, _j, point, isAppend) {
```
Comment (sculove) [inline src/layouts/JustifiedLayout.js]: What is _i. _j ?

File: src/layouts/JustifiedLayout.js [hunk 7]
```diff
@@ -0,0 +1,215 @@
...
+		const style = this.getStyleNames();
+		const _size1 = style[2];
+		const _size2 = style[5];
```
Comment (sculove) [inline src/layouts/JustifiedLayout.js]: number is lowers readability. Could you change meaning const or string?

File: test/unit/JustifiedLayout.spec.js [hunk 8]
```diff
@@ -0,0 +1,190 @@
...
+	const items = [];
+	for (let i = 0; i < 22; ++i) {
+		items.push(getItem(group, i));
+	}
+	return items;
+}
+function round(num) {
+	return parseInt(num * 10000, 10);
+}
+describe("GoogleLayout Test", function() {
```
Comment (sculove) [inline test/unit/JustifiedLayout.spec.js]: Could you change description from GoogleLayout toJustifiedLayout?

File: test/unit/JustifiedLayout.spec.js [hunk 9]
```diff
@@ -0,0 +1,190 @@
...
+	const group = parseInt(Math.random() * 100, 10);
+	const items = [];
+	for (let i = 0; i < 22; ++i) {
+		items.push(getItem(group, i));
+	}
+	return items;
+}
+function round(num) {
+	return parseInt(num * 10000, 10);
+}
```
Comment (sculove) [inline test/unit/JustifiedLayout.spec.js]: How about separating into testUtil file?

File: test/unit/JustifiedLayout.spec.js [hunk 10]
```diff
@@ -0,0 +1,190 @@
...
+				this.inst.width = 800;
+				this.inst.height = 800;
+			});
+			after(() => {
+				if (this.inst) {
+					this.inst = null;
+				}
+			});
+			it("shoud check a append above outline", () => {
+				const items = getItems();
```
Comment (sculove) [inline test/unit/JustifiedLayout.spec.js]: Test code guid is Given/When/Then.
If you possible, I would like to write according to the guide.

Comment (sculove) [reply test/unit/JustifiedLayout.spec.js]: ref https://sculove.github.io/slides/testing/#/25

**Comment by itoolsg** [comment]
skip: apply review

**Pull request merged**